gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJmxMonitor.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    private void processDiscoveryNodeManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                for (JsonNode attribute : attributes) {
+                    if ("ActiveNodeCount".equals(attribute.get("name").asText())) {
+                        int activeNodes = attribute.get("value").asInt();
+                        clusterStats.numWorkerNodes(activeNodes)
+                                .healthy(activeNodes - 1 > 0);
+                        log.debug("Processed DiscoveryNodeManager: ActiveNodeCount = %d, Health = %s",
+                                activeNodes, activeNodes - 1 > 0 ? "Healthy" : "Unhealthy");
+                        break;
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing DiscoveryNodeManager stats");
+        }
+    }
+
+    private void processQueryManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                int queuedQueries = 0;
+                int runningQueries = 0;
+                for (JsonNode attribute : attributes) {
+                    String name = attribute.get("name").asText();
+                    if ("QueuedQueries".equals(name)) {
+                        queuedQueries = attribute.get("value").asInt();
+                    }
+                    else if ("RunningQueries".equals(name)) {
+                        runningQueries = attribute.get("value").asInt();
+                    }
+                }
+                clusterStats.queuedQueryCount(queuedQueries).runningQueryCount(runningQueries);
+                log.debug("Processed QueryManager: QueuedQueries = %d, RunningQueries = %d",
+                        queuedQueries, runningQueries);
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing QueryManager stats");
+        }
+    }
+
+    private String queryJmx(ProxyBackendConfiguration backend, String mbeanName)
+    {
+        String jmxUrl = backend.getProxyTo() + JMX_PATH + "/" + mbeanName;
+        log.debug("Querying JMX at URL: %s", jmxUrl);
+        OkHttpClient client = new OkHttpClient.Builder().build();
+
+        Request request = new Request.Builder()
+                .url(jmxUrl)
+                .addHeader("Authorization", Credentials.basic(username, password))
+                .get()
+                .build();
+
+        Call call = client.newCall(request);
+
+        try (Response res = call.execute()) {
+            if (fromStatusCode(res.code()) == io.airlift.http.client.HttpStatus.OK) {
```

Comment:
it'd be more concise if we can import `io.airlift.http.client.HttpStatus`

Reply:
Done